webrender/src/tiling.rs

```diff
@@ -680,6 +682,29 @@ impl RenderTask {
 pub const SCREEN_TILE_SIZE: i32 = 256;
 pub const RENDERABLE_CACHE_SIZE: i32 = 2048;
 
+#[derive(Clone, Copy, Debug)]
+pub enum MaskImageSource {
+    User(ImageKey),
+    Renderer(TextureId),
+}
+
+/// Per-batch clipping info merged with the mask image.
+#[derive(Clone, Debug)]
+pub struct MaskedClip {
```

Comment:
I can see your point about the consistency aspect of the structure internals, but I think having the public API type for FrameBuilder be called Clip is important. How about as a compromise we keep the structure layout as is, but rename Clip -> ClipGpu and make it private, and rename MaskedClip -> Clip?